src/pages/workspaces/workspace/WorkspaceContainer.js

```diff
@@ -230,11 +230,14 @@ export const wrapWorkspace = ({ breadcrumbs, activeTab, title, topBarContent, sh
     const accessNotificationId = useRef()
     const cachedWorkspace = Utils.useStore(workspaceStore)
     const [loadingWorkspace, setLoadingWorkspace] = useState(false)
-    const { runtimes, refreshRuntimes, persistentDisks, galaxyDataDisks } = useCloudEnvironmentPolling(namespace)
-    const { apps, refreshApps } = useAppPolling(namespace, name)
     const workspace = cachedWorkspace && _.isEqual({ namespace, name }, _.pick(['namespace', 'name'], cachedWorkspace.workspace)) ?
       cachedWorkspace :
       undefined
+    const [googleProject, setGoogleProject] = useState(workspace ? workspace.workspace.googleProject : undefined)
+    const prevGoogleProject = Utils.usePrevious(googleProject)
+    const { runtimes, refreshRuntimes, persistentDisks, galaxyDataDisks } = useCloudEnvironmentPolling(googleProject)
+    const { apps, refreshApps } = useAppPolling(googleProject, name)
+    googleProject !== prevGoogleProject && refreshRuntimes() && refreshApps()
```

Comment:
@dmohs we especially want careful review of this line, because it seems potentially bad if `refreshRuntimes` started returning undefined? Another potential option would be `googleProject !== prevGoogleProject && (refreshRuntimes() || refreshApps())` but that just rubs me the wrong way for some reason? Do you have any thoughts here?

Reply:
maybe just
```js
    if (googleProject !== prevGoogleProject) {
      refreshRuntimes()
      refreshApps()
    }
```